src/injected-js/gmail/sync-compose-processor-20220909.ts

```diff
@@ -14,11 +14,29 @@ export function parseComposeRequestBody_2022_09_09(request: Array<any>) {
   );
 }
 
-export function parseComposeResponseBody_2022_09_09(response: Array<any>) {
-  return (
-    parseUpdateDraftResponseBody(response) ||
-    parseSendDraftResponseBody(response)
-  );
+export function parseComposeResponseBody_2022_09_09(
+  response: Array<any>
+): any[] {
+  const threads = response[1]?.[5];
+
+  return (threads ?? []).flatMap((threadWrapper: unknown) => {
+    if (!Array.isArray(threadWrapper) || !threadWrapper?.[0]) {
+      // exit cuz cannot parse
+      return [];
+    }
+
+    const thread = threadWrapper[0];
+
+    const maybeBody =
+      parseUpdateDraftResponseBody(thread) ||
+      parseSendDraftResponseBody(thread);
+
+    if (maybeBody) {
+      return [maybeBody];
+    }
+
+    return [];
```

Comment:
from this code looks like first thread is being parsed always, but in this gist 
https://gist.github.com/bkorobeinikov/cc1eaf367bd0a5fa84b9beb98a1b42b8#file-gistfile1-txt

second item has correct value based on draftId in the request

Reply:
Can you link to the line you think the draft id should have? The PR as of 85ac92b seems to pick up two message ids in this test https://github.com/InboxSDK/InboxSDK/pull/824/files#diff-383214f3ceee76012312efab301a5a77b3511352e93d08eff206f136045adb0cR64 with the gist linked.

We maybe dealing with trees of data in the batch response instead of just nested arrays.

Reply:
my mistake, there is a wrapper array around thread array, didn't notice it